backend/app/models.py

```diff
@@ -8,40 +8,274 @@
+        Allows the user to iterate over all of the pronouns in this group. Pronouns
+        are returned in lowercase and order is not guaranteed.
+        >>> pronouns = ['she', 'her', 'her', 'hers', 'herself']
+        >>> pronoun_group = PronounSeries.objects.create('Fem', *pronouns)
+        >>> sorted(pronoun_group)
+        ['her', 'hers', 'herself', 'she']
+        """
+
+        all_pronouns = self.get_all_pronouns()
+        yield from all_pronouns
+
+    def __repr__(self):
+        """
+        >>> PronounSeries.objects.create(
+        ...     identifier='Masc',
+        ...     subj='he',
+        ...     obj='him',
+        ...     pos_det='his',
+        ...     pos_pro='his',
+        ...     reflex='himself'
+        ... )
+        <Masc: ['he', 'him', 'himself', 'his']>
+        :return: A console-friendly representation of the pronoun series
+        """
+
+        return f'<{self.identifier}: {list(sorted(self))}>'
+
+    def __str__(self):
+        """
+        >>> str(PronounSeries.objects.create('Andy', *['Xe', 'Xem', 'Xis', 'Xis', 'Xemself']))
+        'Andy-series'
+        :return: A string-representation of the pronoun series
+        """
+
+        return self.identifier + '-series'
+
+    def __hash__(self):
+        """
+        Makes the `PronounSeries` class hashable
+        """
+
+        return self.identifier.__hash__()
+
+    def __eq__(self, other):
+        """
+        Determines whether two `PronounSeries` are equal. Note that they are only equal if
+        they have the same identifier and the exact same set of pronouns.
+
+        >>> fem_series = PronounSeries.create(
+        ...     identifier='Fem',
+        ...     subj='she',
+        ...     obj='her',
+        ...     pos_det='her',
+        ...     pos_pro='hers',
+        ...     reflex='herself'
+        ... )
+        >>> second_fem_series = PronounSeries.create(
+        ...     identifier='Fem',
+        ...     subj='she',
+        ...     obj='her',
+        ...     pos_pro='hers',
+        ...     reflex='herself'
+        ...     pos_det='HER',
+        ... )
+        >>> fem_series == second_fem_series
+        True
+        >>> masc_series = PronounSeries.create(
+        ...     identifier='Masc',
+        ...     subj='he',
+        ...     obj='him',
+        ...     pos_det='his',
+        ...     pos_pro='his',
+        ...     reflex='himself'
+        ... )
+        >>> fem_series == masc_series
+        False
+        :param other: The `PronounSeries` object to compare
+        :return: `True` if the two series are the same, `False` otherwise.
+        """
+
+        return (
+                self.identifier == other.identifier
+                and sorted(self) == sorted(other)
+        )
+
+
+class Gender(models.Model):
+    """
+    This model defines a gender that analysis functions will use to operate.
+    """
+
+    label = models.CharField(max_length=60)
+    pronoun_series = models.ManyToManyField(PronounSeries)
```

Comment:
On further reflection, is `ManyToMany` here correct? Is it possible for a single `Gender` to have multiple `PronounSeries`? I suppose that would allow us to do something like create a binary group gender that references multiple `PronounSeries`.

Reply:
On reading through the commented tests below, I think that this makes sense. I think the commented-out methods below are worthwhile in this PR, then, and we should un-comment them. Otherwise, we have to go through some convolutions to get all of the relevant subject pronouns for a given `Gender`.

Reply:
Since the `PronounSeries` model only allows for 5 identifiers (versus 10), I thought of an example where a user might create a `Gender` instance with a set of singular pronouns and plural pronouns for some analysis.